test/legacy-upgrade/create-in-v0.108.0-sources.td

```diff
@@ -0,0 +1,80 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
```

Comment:
@MaterializeInc/testing I added this file to confirm that the migration I added doesn't break sources/subsources that already exist -- is there any other system I should add a test to to verify the migration?

Reply:
platform-checks would be great, maybe a new check or modifying an existing source: https://github.com/MaterializeInc/materialize/blob/main/doc/developer/platform-checks.md

Reply:
Alright I pushed a check that modifies some existing platform checks to make sure they test migrations of load-gen subsources, and mysql/postgres subsources with text column options.
The `plan_create_subsource` method in `src/sql/src/plan/statement/ddl.rs` will error out if a subsource has not been correctly migrated, so we can be sure that things are working with the `details` option.

There is one commented out `regexp_match` check I added in `misc/python/materialize/checks/all_checks/mysql_cdc.py` but couldn't get the quoting to work correctly (though when verified using `mzcompose sql materialized` and running the query myself it was correct)

If either of you have a second to see if you spot the issue that would be appreciated.